test/ddthost_test/CMakeLists.txt

```diff
@@ -0,0 +1,191 @@
+  ADD_COMPILE_OPTIONS(-warn)
+  ADD_COMPILE_OPTIONS(-traceback)
+  ADD_COMPILE_OPTIONS(-debug extended)
+  ADD_COMPILE_OPTIONS(-fpp)
+elseif (${CMAKE_Fortran_COMPILER_ID} MATCHES "PGI")
+#  pgf90
+#  MESSAGE("pgf90 being used.")
+  ADD_COMPILE_OPTIONS(-g)
+  ADD_COMPILE_OPTIONS(-Mipa=noconst)
+  ADD_COMPILE_OPTIONS(-traceback)
+  ADD_COMPILE_OPTIONS(-Mfree)
+  ADD_COMPILE_OPTIONS(-Mfptrap)
+  ADD_COMPILE_OPTIONS(-Mpreprocess)
+else (${CMAKE_Fortran_COMPILER_ID} MATCHES "GNU")
+  message (WARNING "This program has only been compiled with gfortran, pgf90 and ifort. If another compiler is needed, the appropriate flags SHOULD be added in ${CMAKE_SOURCE_DIR}/CMakeLists.txt")
+endif (${CMAKE_Fortran_COMPILER_ID} MATCHES "GNU")
+
+#------------------------------------------------------------------------------
+# CMake Modules
+# Set the CMake module path
+list(APPEND CMAKE_MODULE_PATH "${CCPP_FRAMEWORK}/cmake")
+#------------------------------------------------------------------------------
+# Set OpenMP flags for C/C++/Fortran
+if (OPENMP)
+  include(detect_openmp)
+  detect_openmp()
+  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
+  set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+  set (CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${OpenMP_Fortran_FLAGS}")
+  message(STATUS "Enable OpenMP support for C/C++/Fortran compiler")
+else(OPENMP)
+  message (STATUS "Disable OpenMP support for C/C++/Fortran compiler")
+endif()
+
+# Create metadata and source file lists
+FOREACH(FILE ${SCHEME_FILES})
+  FILE(STRINGS ${FILE} FILENAMES)
+  LIST(APPEND SCHEME_FILENAMES ${FILENAMES})
+ENDFOREACH(FILE)
+string(REPLACE ";" "," SCHEME_METADATA "${SCHEME_FILES}")
+
+FOREACH(FILE ${SCHEME_FILENAMES})
+  # target_sources prefers absolute pathnames
+  string(REPLACE ".meta" ".F90" TEMP "${FILE}")
+  get_filename_component(ABS_PATH "${TEMP}" ABSOLUTE)
+  list(APPEND LIBRARY_LIST ${ABS_PATH})
+ENDFOREACH(FILE)
+
+FOREACH(FILE ${HOST_FILES})
+  LIST(APPEND HOST_METADATA "${FILE}.meta")
+  # target_sources prefers absolute pathnames
+  get_filename_component(ABS_PATH "${FILE}.F90" ABSOLUTE)
+  LIST(APPEND HOST_SOURCE   "${ABS_PATH}")
+ENDFOREACH(FILE)
+list(APPEND LIBRARY_LIST ${HOST_SOURCE})
+string(REPLACE ";" ".meta," HOST_METADATA "${HOST_FILES}")
+set(HOST_METADATA "${HOST_METADATA}.meta,${HOST}.meta")
+
+string(REPLACE ";" "," SUITE_XML "${SUITE_FILES}")
+
+# Run ccpp_capgen
+set(CAPGEN_CMD "${CCPP_FRAMEWORK}/ccpp_capgen.py")
+list(APPEND CAPGEN_CMD "--host-files")
+list(APPEND CAPGEN_CMD "${HOST_METADATA}")
+list(APPEND CAPGEN_CMD "--scheme-files")
+list(APPEND CAPGEN_CMD "${SCHEME_METADATA}")
+list(APPEND CAPGEN_CMD "--suites")
+list(APPEND CAPGEN_CMD "${SUITE_XML}")
+list(APPEND CAPGEN_CMD "--host-name")
+list(APPEND CAPGEN_CMD "test_host")
+list(APPEND CAPGEN_CMD "--output-root")
+list(APPEND CAPGEN_CMD "${CCPP_CAP_FILES}")
+string(REPEAT "--verbose" ${VERBOSITY} VERBOSE_REPEATED)
+list(APPEND CAPGEN_CMD "--verbose")
```

Comment:
```suggestion
list(APPEND CAPGEN_CMD ${VERBOSE_REPEATED})
```

Reply:
😳
fixed!

Reply:
@mwaxmonsky, thanks for the catch but is there a reason I could not just change line 148 to be:
```
string(REPEAT "--verbose;" ${VERBOSITY} VERBOSE_REPEATED)
```
It leads to an extra space between the last `--verbose` and the `--debug` but bash does not care.

Reply:
Oh, good idea.  I didn't realize CMake interpreted strings like that.  Approved!